descwl_coadd_task/_coadd_task.py

```diff
@@ -0,0 +1,509 @@
+    mfrac_warp : `~lsst.afw.image.ExposureF`
+        The masked fraction warped exposure.
+    exp_info : `dict`
+        A dictionary containing the exposure metadata.
+    """
+
+    warp: afwImage.ExposureF
+    noise_warp: afwImage.ExposureF
+    mfrac_warp: afwImage.ExposureF
+    exp_info: dict[str, Any]
+
+    # Mimimal set of methods to make this duck-type as Exposure objects.
+    def getFilter(self):
+        return self.warp.getFilter()
+
+
+class AssembleShearCoaddSlowConnections(AssembleShearCoaddConnections):
+    pass
+
+
+class AssembleShearCoaddSlowConfig(
+    AssembleShearCoaddConfig, pipelineConnections=AssembleShearCoaddSlowConnections
+):
+    """Configuration for AssembleShearCoaddSlowTask.
+
+    This is a trivial sub-class of `AssembleShearCoaddConfig`.
+    """
+
+
+class AssembleShearCoaddSlowTask(AssembleShearCoaddTask):
+    """A PipelineTask that assembles coadds from warps, one cell at a time.
+
+    The run method of this PipelineTask calls the `make_coadd` function from
+    the `descwl_coadd` package. It constructs one cell completely before moving
+    onto the next one, and thus leads to I/O patterns that are too suboptimal
+    to be used for large-scale production runs. This task is used to check for
+    consistency with `AssembleShearCoaddTask` which uses the barebones from the
+    `descwl_coadd` package.
+    """
+
+    ConfigClass = AssembleShearCoaddSlowConfig
+    _DefaultName = "assembleShearCoaddSlow"
+
+    def run(self, *, input_warps, noise_warps, mfrac_warps, skyInfo):
+        """Coadd a set of warped images, along with noise and masked fractions.
+
+        Parameters
+        ----------
+        input_warps : `Iterable` [`~lsst.daf.butler.DeferredDatasetHandle`]
+            List of data references to warped images.
+        noise_warps : `Iterable` [`~lsst.daf.butler.DeferredDatasetHandle`]
+            List of data references of warped noise images.
+        mfrac_warps : `Iterable` [`~lsst.daf.butler.DeferredDatasetHandle`]
+            List of data references of masked fraction images.
+        skyInfo : `~lsst.pipe.base.Struct`
+            A Struct object
+
+        Returns
+        -------
+        coaddExposure : `~lsst.afw.image.ExposureF`
+            The coadded exposure.
+        """
+        # statsCtrl = self._construct_stats_control()
+
+        # gc = self._construct_grid_container(skyInfo, statsCtrl)
+        # coadd_inputs_gc = GridContainer(gc.shape)
+        edge = afwImage.Mask.getPlaneBitMask("EDGE")
+
+        cells: list[SingleCellCoadd] = []
+        for cellInfo in skyInfo.patchInfo:
+            # coadd_inputs = self.inputRecorder.makeCoaddInputs()
+            # # Reserve the absolute maximum of how many ccds, visits
+            # # we could potentially have.
+            # coadd_inputs.ccds.reserve(len(input_warps))
+            # coadd_inputs.visits.reserve(len(input_warps))
+            # coadd_inputs_gc[cellInfo.index] = coadd_inputs
```

Comment:
What is going on with this commented out block. Can you clarify with more comments?

Reply:
Yeah, I should remove these lines. This is from an earlier approach to implementing this but is no longer relevant.